src/app/portlets/dot-edit-content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.ts

```diff
@@ -1,22 +1,94 @@
-import { Component, OnInit, Input, EventEmitter, Output } from '@angular/core';
+import { Component, OnInit, Input, EventEmitter, Output, ViewChild } from '@angular/core';
 import { DotMessageService } from '../../../../api/services/dot-messages-service';
+import { SelectItem, MenuItem, SplitButton, InputSwitch } from 'primeng/primeng';
+import { Workflow } from '../../../../shared/models/workflow/workflow.model';
 
 @Component({
     selector: 'dot-edit-page-toolbar',
     templateUrl: './dot-edit-page-toolbar.component.html',
     styleUrls: ['./dot-edit-page-toolbar.component.scss']
 })
 export class DotEditPageToolbarComponent implements OnInit {
-    @Input() pageTitle: string;
+    @ViewChild('actionsButton') actionsButton: SplitButton;
+    @ViewChild('locker') locker: InputSwitch;
+
     @Input() canSave: boolean;
+    @Input() pageLocked: boolean;
+    @Input() pageTitle: string;
+    @Input() pageUrl: string;
+    @Input() pageWorkflows: Workflow[];
 
+    @Output() lockPage = new EventEmitter<boolean>();
+    @Output() pageState = new EventEmitter<string>();
     @Output() save = new EventEmitter<MouseEvent>();
 
+    states: SelectItem[] = [];
+    stateSelected = 'preview';
+    workflowsActions: MenuItem[] = [];
+
     constructor(public dotMessageService: DotMessageService) {}
 
     ngOnInit() {
-        this.dotMessageService.getMessages([
-            'editpage.toolbar.primary.action'
-        ]).subscribe();
+        this.dotMessageService
+            .getMessages([
+                'editpage.toolbar.primary.action',
+                'editpage.toolbar.edit.page',
```

Comment:
what do you think on add a common label that has the 'Edit' text, it looks like a pretty common text across the app.

Reply:
Honestly don't know at this point, need to come up with something for the whole messages keys.